proxy-node-gateway/src/main/java/uk/gov/ida/notification/resources/EidasAuthnRequestResourceV2.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.ida.notification.resources;
+
+import io.dropwizard.jersey.sessions.Session;
+import io.dropwizard.views.View;
+import org.apache.commons.lang.StringUtils;
+import org.opensaml.saml.saml2.ecp.RelayState;
+import uk.gov.ida.notification.EidasSamlParserService;
+import uk.gov.ida.notification.SamlFormViewBuilder;
+import uk.gov.ida.notification.VSPService;
+import uk.gov.ida.notification.dto.EidasSamlParserRequest;
+import uk.gov.ida.notification.dto.EidasSamlParserResponse;
+import uk.gov.ida.notification.saml.SamlFormMessageType;
+import uk.gov.ida.notification.dto.VSPAuthnRequestResponse;
+import uk.gov.ida.notification.views.SamlFormView;
+
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.FormParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.MediaType;
+import java.net.URI;
+import java.util.logging.Logger;
+
+@Path("/SAML2/SSO")
+public class EidasAuthnRequestResourceV2 {
+
+    private final Logger log = Logger.getLogger(getClass().getName());
+    public static final String SESSION_KEY_EIDAS_REQUEST_ID = "eidas_request_id";
+    public static final String SESSION_KEY_EIDAS_CONNECTOR_PUBLIC_KEY = "eidas_connector_public_key";
+    public static final String SESSION_KEY_EIDAS_DESTINATION = "eidas_destination";
+    public static final String SESSION_KEY_HUB_REQUEST_ID = "hub_request_id";
+    public static final String SUBMIT_BUTTON_TEXT = "Post Verify Authn Request to Hub";
+
+    private final EidasSamlParserService eidasSamlParserService;
+    private final VSPService vspService;
+    private final SamlFormViewBuilder samlFormViewBuilder;
+
+    public EidasAuthnRequestResourceV2(
+            EidasSamlParserService eidasSamlParserService,
+            VSPService vspService,
+            SamlFormViewBuilder samlFormViewBuilder) {
+        this.eidasSamlParserService = eidasSamlParserService;
+        this.vspService = vspService;
+        this.samlFormViewBuilder = samlFormViewBuilder;
+    }
+
+    @GET
+    @Path("/Redirect")
+    public View handleRedirectBinding(
+            @QueryParam(SamlFormMessageType.SAML_REQUEST) String encodedEidasAuthnRequest,
+            @QueryParam("RelayState") String relayState,
+            @Session HttpSession session) {
+        return handlePostBinding(encodedEidasAuthnRequest, relayState, session);
```

Comment:
Could this call `handleAuthnRequest` instead of `handlePostBinding?`. Save the extra method call.

Reply:
The idea was to emphasise that these resource were equivalent, but yes think it makes sense to just call handleAuthnRequest directly.